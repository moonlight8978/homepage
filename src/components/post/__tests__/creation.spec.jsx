import React from 'react'
import { render } from '@testing-library/react'

import Creation from '../creation'

test('renders formatted modified time', () => {
  const updatedAt = Date.parse(
    'Mon Jan 13 2020 09:00:00 GMT+0700 (Indochina Time)'
  )
  const { container, getByText } = render(
    <Creation updatedAt={updatedAt} author="Sample" />
  )
  expect(getByText('January 13, 2020')).toBeInTheDocument()
  expect(container.firstChild).toMatchSnapshot()
})

test('renders formatted modified time with padding', () => {
  const updatedAt = Date.parse(
    'Mon Jan 02 2020 09:00:00 GMT+0700 (Indochina Time)'
  )
  const { container, getByText } = render(
    <Creation updatedAt={updatedAt} author="Sample" />
  )
  expect(getByText('January 02, 2020')).toBeInTheDocument()
  expect(container.firstChild).toMatchSnapshot()
})

test('renders author correctly', () => {
  const author = '_Moon_'
  const { container, getByText } = render(
    <Creation updatedAt={0} author={author} />
  )
  expect(getByText('by _Moon_')).toBeInTheDocument()
  expect(container.firstChild).toMatchSnapshot()
})

test('match snapshot', () => {
  const updatedAt = Date.parse(
    'Mon Jan 13 2020 09:00:00 GMT+0700 (Indochina Time)'
  )
  const author = '_Moon_'
  const { container } = render(
    <Creation updatedAt={updatedAt} author={author} />
  )
  expect(container.firstChild).toMatchSnapshot()
})
