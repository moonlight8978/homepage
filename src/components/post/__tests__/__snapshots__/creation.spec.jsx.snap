// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<span
  class="creation"
>
  <span>
    January 13, 2020
  </span>
  <span>
     by _Moon_
  </span>
</span>
`;

exports[`renders author correctly 1`] = `
<span
  class="creation"
>
  <span>
    January 01, 1970
  </span>
  <span>
     by _Moon_
  </span>
</span>
`;

exports[`renders formatted modified time 1`] = `
<span
  class="creation"
>
  <span>
    January 13, 2020
  </span>
  <span>
     by Sample
  </span>
</span>
`;

exports[`renders formatted modified time with padding 1`] = `
<span
  class="creation"
>
  <span>
    January 02, 2020
  </span>
  <span>
     by Sample
  </span>
</span>
`;
